functions-app/lib/mtc-fs-utils.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const fs = require('fs-extra')
+const os = require('os')
+const path = require('path')
+
+const base = require('./logger')
+const functionName = 'mtc-fs-utils'
+
+const mtcFsUtils = {
+  /**
+   * Work out if we can use the supplied path as a writeable directory.  If not, use os.tmpdir
+   * @param {String} path - the directory we want to write to
+   * @return {Promise<(() => string)|*>}
```

Comment:
is that valid jsdoc syntax?
usually the functions are described like `{function(): string}`

Reply:
Good spot - we returning a string, not a function